servers/src/main/java/tachyon/master/file/meta/InodeTree.java

```diff
@@ -240,6 +240,13 @@ public CreatePathResult createPath(TachyonURI path, CreatePathOptions options)
     InodeDirectory currentInodeDirectory = (InodeDirectory) traversalResult.getInode();
     List<Inode> createdInodes = Lists.newArrayList();
     List<Inode> modifiedInodes = Lists.newArrayList();
+    // Directory persistence will not happen until the end of this method. So we take notes of the
+    // following two kinds of inodes to be persisted:
+    // (1) The existing inodes that need to be persisted.
+    List<Inode> toBePersistedOldInodes = traversalResult.getToBePersisted();
```

Comment:
Why do we need to distinguish between the existing and newly created inodes here? It seems like you could just use a single `inodesToPersist` list.